### DISCUSSION TITLE
Static site generation with Vue causes SyntaxError: Cannot use import statement outside a module

### DISCUSSION THREAD

**Discussion by liborjelinek** [vikejs/vike#1153] [Discussion] [answered]
Has anybody already integrate PrimeVue (https://primevue.org) with Vike for SSG? My app works works great on "npm run dev", but for "npm run build" it causes `SyntaxError: Cannot use import statement outside a module`.

For example, when I try to generate page containing Button:

```
...
vike v0.4.142 pre-rendering HTML...
(node:65063) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.
(Use `node --trace-warnings ...` to show where the warning was created)
/Users/matt/git-dm/www/.my/blank-vue/node_modules/primevue/button/button.esm.js:1
import Badge from 'primevue/badge';
^^^^^^

SyntaxError: Cannot use import statement outside a module
```

**Comment by brillout** [comment]
It's a classic CJS/ESM issue and there are usually multiple ways to work around it.

Create a minimal reproduction (that doesn't use pre-rendering) and I'll dig into it.

**Comment by liborjelinek** [comment]
I will prepare it! Thank you so much.

In the meanwhile: how it is even possible that the same code is behaving differently on npm run and build?

**Reply by brillout** [reply]
See comment at the end of the readme of my fork of your reproduction.

**Comment by liborjelinek** [comment]
Here you go https://github.com/mattwarrick/vike-primevue. `npm run dev` is okay, the error appears only on `npm run build` (Vike instructed to prerender).

Specifically, have a look if I [add PrimeVue](https://github.com/mattwarrick/vike-primevue/commit/1019dba12c578cae8007ef9cb7eb171aad36570c) correctly.

Thank you so much!

**Reply by brillout** [reply, marked as answer]
Improved reproduction: https://github.com/brillout/vike-primevue.

The workaround is [this](https://github.com/brillout/vike-primevue/commit/dc5b5d4d7356a003468709dca35968944978f5bd).

**Reply by brillout** [reply]
Btw. would your company be up for [sonsoring](https://github.com/sponsors/brillout)?

**Reply by liborjelinek** [reply]
Thank you! I will never ever figure it out myself. It is too much black magic.

Ad sponsoring. I am just evaluating Vue.js SSGs area for side project of small website with a handful of marking pages only. Vike looks like powerful full-fledged SSR. But maybe I look for more Vue and SSG centered tool. Leverage existing Vue.js with Vue Router and produce static HTML files. (I dislike Nuxt since it is very slow and black box.)

**Reply by brillout** [reply]
Makes sense 👍 

> I will never ever figure it out myself. It is too much black magic.

I agree; it's quite an unfortunate situation that many packages publish wrong ESM/CJS code. (FYI I'm pushing the Vite team to automatically apply workarounds such as `ssr.noExternal`, so that everything works seamlessly for the user.)
